src/engraving/dom/noteline.cpp

```diff
@@ -0,0 +1,250 @@
+{
+    if (pid == Pid::OFFSET) {
+        return Sid::NOSTYLE;
+    }
+    return TextLineBase::getPropertyStyle(pid);
+}
+
+NoteLine::NoteLine(EngravingItem* parent)
+    : TextLineBase(ElementType::NOTELINE, parent, ElementFlag::MOVABLE)
+{
+    setAnchor(Spanner::Anchor::NOTE);
+    setDiagonal(true);
+    setSystemFlag(false);
+    setPlacement(PlacementV::ABOVE);
+
+    initElementStyle(&noteLineStyle);
+
+    setBeginText(u"");
+    setContinueText(u"");
+    setEndText(u"");
+    setBeginTextOffset(PointF(0, 0));
+    setContinueTextOffset(PointF(0, 0));
+    setEndTextOffset(PointF(0, 0));
+    setLineVisible(true);
+
+    setBeginHookType(HookType::NONE);
+    setEndHookType(HookType::NONE);
+    setBeginHookHeight(Spatium(1.5));
+    setEndHookHeight(Spatium(1.5));
+    setGapBetweenTextAndLine(Spatium(0.5));
+
+    setLineEndPlacement(NoteLineEndPlacement::OFFSET_ENDS);
+
+    resetProperty(Pid::BEGIN_TEXT_PLACE);
+    resetProperty(Pid::CONTINUE_TEXT_PLACE);
+    resetProperty(Pid::END_TEXT_PLACE);
+}
+
+NoteLine::NoteLine(const NoteLine& nl)
+    : TextLineBase(nl)
+{
+}
+
+LineSegment* NoteLine::createLineSegment(System* parent)
+{
+    NoteLineSegment* seg = new NoteLineSegment(this, parent);
+    seg->setTrack(track());
+    return seg;
+}
+
+PropertyValue NoteLine::propertyDefault(Pid propertyId) const
+{
+    switch (propertyId) {
+    case Pid::BEGIN_TEXT:
+    case Pid::CONTINUE_TEXT:
+    case Pid::END_TEXT:
+        return "";
+    case Pid::LINE_VISIBLE:
+        return true;
+    case Pid::BEGIN_TEXT_OFFSET:
+    case Pid::CONTINUE_TEXT_OFFSET:
+    case Pid::END_TEXT_OFFSET:
+        return PointF(0, 0);
+    case Pid::BEGIN_HOOK_TYPE:
+    case Pid::END_HOOK_TYPE:
+        return HookType::NONE;
+    case Pid::BEGIN_TEXT_PLACE:
+    case Pid::CONTINUE_TEXT_PLACE:
+    case Pid::END_TEXT_PLACE:
+        return TextPlace::LEFT;
+    case Pid::BEGIN_HOOK_HEIGHT:
+    case Pid::END_HOOK_HEIGHT:
+        return Spatium(1.5);
+    case Pid::PLACEMENT:
+        return PlacementV::ABOVE;
+    case Pid::DIAGONAL:
+        return true;
+    case Pid::NOTELINE_PLACEMENT:
+        return NoteLineEndPlacement::OFFSET_ENDS;
+    case Pid::OFFSET:
+        return PointF();
+    default:
+        return TextLineBase::propertyDefault(propertyId);
+    }
+}
+
+PropertyValue NoteLine::getProperty(Pid propertyId) const
+{
+    switch (propertyId) {
+    case Pid::PLACEMENT:
+        return PlacementV::ABOVE;
+    case Pid::NOTELINE_PLACEMENT:
+        return m_lineEndPlacement;
+    default:
+        return TextLineBase::getProperty(propertyId);
+    }
+}
+
+bool NoteLine::setProperty(Pid propertyId, const PropertyValue& val)
+{
+    switch (propertyId) {
+    case Pid::NOTELINE_PLACEMENT:
+        setLineEndPlacement(val.value<NoteLineEndPlacement>());
+        break;
+    default:
+        return TextLineBase::setProperty(propertyId, val);
+    }
+
+    return true;
+}
+
+void NoteLine::addLineAttachPoints()
```

Comment:
This might be beyond the scope of your PR, but if you have the time/patience: this function dates back to before we had the layout modules, but I think it should go there now as it's basically layout code (lineAttachPoints are the means by which the horizontal spacing system knows to make space for the minimum length of these lines). Perhaps this could become `ChordLayout::addLineAttachPoints(NoteLine* noteLine)`?

Reply:
Good idea.  I have added a generic one in `ChordLayout` for note anchored lines and glissandos, and also moved `TieSegment::addLineAttachPoints` to its own static method in `SlurTieLayout`.